Source/WebCore/Modules/mediasource/SourceBuffer.cpp

```diff
@@ -1356,11 +1356,23 @@ void SourceBuffer::setShouldGenerateTimestamps(bool flag)
     m_private->setShouldGenerateTimestamps(flag);
 }
 
-void SourceBuffer::sourceBufferPrivateBufferedChanged(const PlatformTimeRanges&, CompletionHandler<void()>&& completionHandler)
+void SourceBuffer::sourceBufferPrivateBufferedChanged(const PlatformTimeRanges& ranges, CompletionHandler<void()>&& completionHandler)
 {
+    ASSERT(ranges != m_buffered->ranges(), "sourceBufferPrivateBufferedChanged should only be called if the ranges did change");
+    if (isManaged()) {
+        auto addedRanges = ranges;
+        addedRanges -= m_buffered->ranges();
+        auto addedTimeRanges = addedRanges.length() ? RefPtr { TimeRanges::create(WTFMove(addedRanges)) } : nullptr;
+
+        auto removedRanges = m_buffered->ranges();
+        removedRanges -= ranges;
+        auto removedTimeRanges = removedRanges.length() ? RefPtr { TimeRanges::create(WTFMove(removedRanges)) } : nullptr;
+
+        ASSERT(addedTimeRanges || removedTimeRanges, "Can't generate an empty dictionary");
+        queueTaskToDispatchEvent(*this, TaskSource::MediaElement, BufferedChangeEvent::create(WTFMove(addedTimeRanges), WTFMove(removedTimeRanges)));
+    }
+    m_buffered = TimeRanges::create(ranges);
```

Comment:
The usual pattern is to update the value in the task queue that fires the event.
Should we update m_buffered just before firing the buffered change event?